packages/back-end/src/models/SSOConnectionModel.ts

```diff
@@ -30,22 +30,28 @@ const SSOConnectionModel = mongoose.model<SSOConnectionDocument>(
   ssoConnectionSchema
 );
 
+function toInterface(doc: SSOConnectionDocument): SSOConnectionInterface {
+  return doc.toJSON();
+}
+
 export async function getSSOConnectionById(
   id: string
 ): Promise<null | SSOConnectionInterface> {
   if (!id) return null;
   const doc = await SSOConnectionModel.findOne({ id });
 
-  return doc ? doc.toJSON() : null;
+  return doc ? toInterface(doc) : null;
 }
 
 export async function getSSOConnectionByEmailDomain(
   emailDomain: string
 ): Promise<null | SSOConnectionInterface> {
   if (!emailDomain) return null;
-  const doc = await SSOConnectionModel.findOne({ emailDomain });
+  const doc = await SSOConnectionModel.findOne({
+    emailDomains: { $in: [emailDomain] },
```

Comment:
Don't need `$in` here.  If the field you're querying is an array, Mongo automatically does an "in" operation behind the scenes.

```suggestion
    emailDomains: emailDomain,
```